### PR TITLE
New Module: Write Windows event log entries (win_eventlog_entry)

### PR DESCRIPTION
**Pull request by andrewsaraceni**
##### SUMMARY
As discussed in https://github.com/ansible/ansible/issues/26825, a module does not exist for the creation of event log entries themselves in Windows event logs.  Rounding out the functionality provided by the `*-EventLog` PowerShell cmdlets along with PR https://github.com/ansible/ansible/pull/27827, this module allows for the generation of entries to event logs from a specified source.

Integration tests and check-mode support are both included.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/windows/win_eventlog_entry.ps1

##### ANSIBLE VERSION

```
ansible 2.4.0
```

##### ADDITIONAL INFORMATION

```yml
- name: Write an entry to a Windows event log
  win_eventlog_entry:
    log: MyNewLog
    source: NewLogSource1
    event_id: 1234
    message: This is a test log entry.

- name: Write another entry to a different Windows event log
  win_eventlog_entry:
    log: AnotherLog
    source: MyAppSource
    event_id: 5000
    message: An error has occurred.
    entry_type: Error
    category: 5
    raw_data: 10,20
```

### CODE REVIEW

File: lib/ansible/modules/windows/win_eventlog_entry.ps1 [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#!powershell
+
+# (c) 2017, Andrew Saraceni <andrew.saraceni@gmail.com>
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_eventlog_entry.ps1:3]: The copyright is a lot simpler now, you only need a few lines. See http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#copyright for what has changed.

File: lib/ansible/modules/windows/win_eventlog_entry.ps1 [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_eventlog_entry.ps1]: You can get right rid of `# WANT_JSON` and `# POWERSHELL_COMMON` and replace it with just `#Requires -Module Ansible.ModuleUtils.Legacy.psm1`. This has more info as well http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general_windows.html#windows-new-module-development.

File: lib/ansible/modules/windows/win_eventlog_entry.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Andrew Saraceni <andrew.saraceni@gmail.com>
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_eventlog_entry.py]: Same thing about the copyright.

File: lib/ansible/modules/windows/win_eventlog_entry.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      - FailureAudit
+      - Information
+      - SuccessAudit
+      - Warning
+  category:
+    description:
+      - A numeric task category associated with the category message file for the log source.
+  raw_data:
+    description:
+      - Binary data associated with the log entry.
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_eventlog_entry.py:49]: Are you able to give a bit more info, seems like it's just a byte array represented by hex values but would be good to see that in the documentation.

**Review by jborean93 — CHANGES_REQUESTED** [review]
Looks good to me, nice work, only a few formatting things to change. One thing you may want to look at is creating a custom test module to return all the info of an event log instead of running 4 shell scripts but that's up to you. I did something similar with `win_security_policy` as getting the actual results was too hard to put into one command https://github.com/ansible/ansible/blob/devel/test/integration/targets/win_security_policy/library/test_win_security_policy.ps1

**Comment by andrewsaraceni** [comment]
Thanks @jborean93, I'll look into making these changes later tonight/tomorrow.

**Comment by andrewsaraceni** [comment]
@jborean93  Just made the changes you requested - let me know if there's anything I missed or that still needs work, and thanks for the idea about the test module!

**Review by jborean93 — APPROVED** [review]
Cool, looks good.

**Comment by jborean93** [comment]
shipit

**Pull request merged**